### ISSUE TITLE
DevTools is slow with Tridactyl being enabled

### ISSUE THREAD

**Issue by maxigaz** [tridactyl/tridactyl#1081]
*   Brief description of the problem: The CSS rules are displayed inside DevTools with a 2-3 seconds long delay. This doesn't happen at all when Tridactyl is disabled.

*   Steps to reproduce:

    1. Open any webpage.
	 2. Bring up DevTools with `Ctrl+Shift+I` or inspect an element with `Ctrl+Shift+C`.
	 3. The CSS rules for the current page (or an inspected element) are shown with a 2-3 seconds long delay.

*   Tridactyl version (`:version`): 1.14.0-1.14.0

*   Firefox version (Top right menu > Help > About Firefox): 62.0.3 (64 bit)

*   URL of the website the bug happens on: https://github.com/tridactyl/tridactyl/issues (or any valid URL)

*   Config (in a new tab, run `:viewconfig`, copy the url and paste it somewhere like gist.github.com): https://privatebin.net/?4303b5c4fbdd1f1e#lZqekkIwid1eeU1855nPQ5M43PfxLUqoj5aesY8fo+U=

*   Contents of ~/.tridactylrc or ~/.config/tridactyl/tridactylrc (if they exist):

```
" FOR VIM
" vim:fdm=marker
" vim: set filetype=vim:

" NOTE FROM DEVS{{{

" Move this to $XDG_CONFIG_DIR/tridactyl/tridactylrc (that's
" ~/.config/tridactyl/tridactylrc to mere mortals) or ~/.tridactylrc and
" install the native messenger (:installnative in Tridactyl). Run :source to
" get it in the browser, or just restart.

" NB: If you want "vim-like" behaviour where removing a line from 
" here makes the setting disappear, uncomment the line below.

" sanitise tridactyllocal tridactylsync

" "}}}

" BINDS{{{

" Basic navigation

bind <A-l> tabnext
bind <A-h> tabprev

" Show bookmarks only

bind <a-o> fillcmdline bmarks

" "}}}
" EXTERNAL EDITOR{{{

set editorcmd konsole -e nvim
" "}}}
" BEHAVIOUR{{{
set homepages [https://searx.ch]
" "}}}
" APPEARANCE{{{
set theme dark
" guiset hoverlink right
" "}}}
```

**Comment by maxigaz** [comment]
Yes, I forgot to mention that my machine can be categorised as low-end nowadays. (It has a Dual Core Intel Celeron G1610 running at 2.6 GHz.)

**Comment by maxigaz** [comment]
The issue is still present on Tridactyl 1.16.1pre2946-cc77de0 and Firefox 67.0.4 (64 bit).

**Comment by Schabolon** [comment]
> 
> 
> The issue is still present on Tridactyl 1.16.1pre2946-cc77de0 and Firefox 67.0.4 (64 bit).

I have experienced a similar problem with Tridactyl 1.16.2pre2999-95db0a8 and Firefox 68.0 (64-bit)

**Comment by maxigaz** [comment]
Today I found something I hadn't noticed before: If I switch to the default colour scheme (with `colourscheme default`), Firefox becomes more responsive when inspecting elements. Still not as snappy as when Tridactyl is disabled completely, but definitely better. When I switch back with `colourscheme dark`, Firefox becomes slow again.

Also, when I inspect something (using the dark colour scheme) and scroll down in the list of applied CSS  rules, I can see that rules coming from Tridactyl take up *a lot* of space in the list. Isn't it possible that this complexity is what slows down Firefox?

(I’m on Firefox 70.0.1 and Tridactyl 1.17.1pre3304-7dcf183.)

**Comment by bovine3dom** [comment]
https://github.com/tridactyl/tridactyl/pull/1990

This has been posited as a fix for non-default colour schemes. It mildly breaks something else so we wanted to know if it did fix it before merging it. Could you try it?

**Comment by maxigaz** [comment]
I’ve just tried that branch. (I used `git clone --single-branch -b arix_mitigate_1834 https://github.com/tridactyl/tridactyl.git` to clone it. I hope that's the right command.)

I don’t notice any change in the DevTools.

**Comment by daveriedstra** [comment]
I think I'm hitting this. Doing anything with CSS in devtools has a 1-2s lag. This includes changing styles and inspecting elements. My computer is squarely a mid-range machine though, a ThinkPad T580 with an i5 processor. Setting `colorscheme default` makes no difference. Disabling Tridactyl fixes the issue.

(Other than the colorscheme thing this seems like the same issue as the original. Let me know if you'd like me to file a new one.)

`:version` - 1.22.1
firefox  99.0

[`:viewconfig` json](https://gitlab.com/-/snippets/2295012)

tridactylrc:
```
" vim: filetype=vim

" Ctrl-F should use the browser's native 'find' functionality.
unbind <C-f>

" But also support Tridactyl search too.
bind / fillcmdline find
bind ? fillcmdline find -?
bind n findnext 1
bind N findnext -1
" Remove search highlighting.
bind <Space>, nohlsearch
```

**Comment by daveriedstra** [comment]
After working with this for longer, I've realized that the delay is in loading the relevant styles for a selected page element. For example:

* Load a page in a new tab
* Open the inspector and selector with Ctrl/Cmd + Shift + C
* Hover various elements with the cursor
  * (Slightly increased latency in the highlight, feels like it's playing catch-up with the cursor)
* Click to select an element
  * It takes a few seconds for that element's styles to load in the inspector panel

Similar behaviour when inspecting through the page's markup in the inspector panel.

I don't know if that narrows it down at all. Let me know if I can provide any more info.
